### ISSUE TITLE
New  HF datasets version causes unit test failures  when using  filtering_lambda

### ISSUE THREAD

**Issue by yoavkatz** [IBM/unitxt#912]
Localized an temporarily disabled test in test_loaders.

@unittest.skip("Currently this fails from datasets 2.20")
    def test_load_from_HF_multiple_innvocation_with_filter(self):
        loader = LoadHF(
            path="CohereForAI/aya_evaluation_suite",
            name="aya_human_annotated",
            filtering_lambda='lambda instance: instance["language"]=="eng"',
        )
        ms = loader.process()
        dataset = ms.to_dataset()
        self.assertEqual(
            list(dataset.keys()), ["test"]
        )  # that HF dataset only has the 'test' split
        self.assertEqual(dataset["test"][0]["language"], "eng")

        ms = loader.process()
        dataset = ms.to_dataset()
        self.assertEqual(
            list(dataset.keys()), ["test"]
        )  # that HF dataset only has the 'test' split
        self.assertEqual(dataset["test"][0]["language"], "eng")

**Comment by dafnapension** [comment]
Hi, I am not sure I got the issue right, so I commented out line 210:
`# @unittest.skip("Currently this fails from datasets 2.20")`
and yet, the whole of `test_loaders.py` completed successfully:

```
(virtual39) dafna@DESKTOP-GM8R3J7:~/workspaces/unitxt$ git diff
diff --git a/tests/library/test_loaders.py b/tests/library/test_loaders.py
index ebc586c3..7b71c513 100644
--- a/tests/library/test_loaders.py
+++ b/tests/library/test_loaders.py
@@ -207,7 +207,7 @@ class TestLoaders(UnitxtTestCase):
         )  # that HF dataset only has the 'test' split
         self.assertEqual(dataset["test"][0]["language"], "arb")

-    @unittest.skip("Currently this fails from datasets 2.20")
+    # @unittest.skip("Currently this fails from datasets 2.20")
     def test_load_from_HF_multiple_innvocation_with_filter(self):
         loader = LoadHF(
             path="CohereForAI/aya_evaluation_suite",
(virtual39) dafna@DESKTOP-GM8R3J7:~/workspaces/unitxt$ python -m unittest tests.library.test_loaders
[Unitxt|INFO|loaders.py:123] 2024-06-23 10:18:02,115 >> LoadCSV sets 'data_classification_policy' to ['proprietary'] by default when loading from local files.
To use a different value or remove this message, explicitly set the `data_classification_policy` attribute of the loader.

/home/dafna/workspaces/unitxt/src/unitxt/dataclass.py:336: DeprecationWarning: Field 'use_query' is deprecated. From now on, default behavior is compatible to use_query=True. Please remove this field from your code.
  original_init(self, *args, **kwargs)
[Unitxt|INFO|loaders.py:98] 2024-06-23 10:18:02,120 >>
Loading limited to 300 instances by setting unitxt.settings.global_loader_limit;
/home/dafna/workspaces/unitxt/src/unitxt/loaders.py:337: ResourceWarning: unclosed file <_io.TextIOWrapper name='/tmp/tmpclfep0sm/train.csv' mode='r' encoding='utf-8'>
  yield row.to_dict()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
[Unitxt|INFO|loaders.py:98] 2024-06-23 10:18:02,129 >>
Loading limited to 300 instances by setting unitxt.settings.global_loader_limit;
/home/dafna/workspaces/unitxt/src/unitxt/loaders.py:337: ResourceWarning: unclosed file <_io.TextIOWrapper name='/tmp/tmpclfep0sm/test.csv' mode='r' encoding='utf-8'>
  yield row.to_dict()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
./home/dafna/workspaces/unitxt/src/unitxt/dataclass.py:336: DeprecationWarning: Field 'use_query' is deprecated. From now on, default behavior is compatible to use_query=True. Please remove this field from your code.
  original_init(self, *args, **kwargs)
[Unitxt|INFO|loaders.py:98] 2024-06-23 10:18:02,141 >>
Loading limited to 300 instances by setting unitxt.settings.global_loader_limit;
/home/dafna/workspaces/unitxt/src/unitxt/loaders.py:337: ResourceWarning: unclosed file <_io.TextIOWrapper name='/tmp/tmp23haxk71/train.tsv' mode='r' encoding='utf-8'>
  yield row.to_dict()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
[Unitxt|INFO|loaders.py:98] 2024-06-23 10:18:02,148 >>
Loading limited to 300 instances by setting unitxt.settings.global_loader_limit;
/home/dafna/workspaces/unitxt/src/unitxt/loaders.py:337: ResourceWarning: unclosed file <_io.TextIOWrapper name='/tmp/tmp23haxk71/test.tsv' mode='r' encoding='utf-8'>
  yield row.to_dict()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
.[Unitxt|INFO|loaders.py:123] 2024-06-23 10:18:02,155 >> LoadHF sets 'data_classification_policy' to ['public'] by default when loading from Huggingface hub.
To use a different value or remove this message, explicitly set the `data_classification_policy` attribute of the loader.

[Unitxt|INFO|loaders.py:98] 2024-06-23 10:18:10,730 >>
Loading limited to 10 instances by setting LoadHF.loader_limit;
/home/dafna/workspaces/unitxt/src/unitxt/dataclass.py:336: DeprecationWarning: Field 'use_query' is deprecated. From now on, default behavior is compatible to use_query=True. Please remove this field from your code.
  original_init(self, *args, **kwargs)
Generating train split: 10 examples [00:01,  5.23 examples/s]
Generating train split: 10 examples [00:01,  9.53 examples/s]
Generating train split: 10 examples [00:01,  9.19 examples/s]
.[Unitxt|INFO|loaders.py:123] 2024-06-23 10:18:14,994 >> LoadHF sets 'data_classification_policy' to ['public'] by default when loading from Huggingface hub.
To use a different value or remove this message, explicitly set the `data_classification_policy` attribute of the loader.

[Unitxt|INFO|loaders.py:98] 2024-06-23 10:18:18,253 >>
Loading limited to 300 instances by setting unitxt.settings.global_loader_limit;
Generating train split: 300 examples [00:02, 145.50 examples/s]
Generating train split: 300 examples [00:01, 291.45 examples/s]
Generating train split: 300 examples [00:01, 256.42 examples/s]
Generating train split: 300 examples [00:03, 84.64 examples/s]
Generating train split: 300 examples [00:00, 302.95 examples/s]
Generating train split: 300 examples [00:00, 316.01 examples/s]
.[Unitxt|INFO|loaders.py:123] 2024-06-23 10:18:28,542 >> LoadHF sets 'data_classification_policy' to ['public'] by default when loading from Huggingface hub.
To use a different value or remove this message, explicitly set the `data_classification_policy` attribute of the loader.

[Unitxt|INFO|loaders.py:98] 2024-06-23 10:18:31,492 >>
Loading limited to 300 instances by setting unitxt.settings.global_loader_limit;
Generating train split: 300 examples [00:01, 170.01 examples/s]
Generating train split: 300 examples [00:01, 160.86 examples/s]
.[Unitxt|INFO|loaders.py:123] 2024-06-23 10:18:35,295 >> LoadHF sets 'data_classification_policy' to ['public'] by default when loading from Huggingface hub.
To use a different value or remove this message, explicitly set the `data_classification_policy` attribute of the loader.

[Unitxt|INFO|loaders.py:185] 2024-06-23 10:18:38,555 >>
Loading filtered by: lambda instance: instance["language"]=="eng";
[Unitxt|INFO|loaders.py:98] 2024-06-23 10:18:38,556 >>
Loading limited to 300 instances by setting unitxt.settings.global_loader_limit;
Generating train split: 250 examples [00:00, 259.29 examples/s]
.[Unitxt|INFO|loaders.py:123] 2024-06-23 10:18:39,587 >> LoadHF sets 'data_classification_policy' to ['public'] by default when loading from Huggingface hub.
To use a different value or remove this message, explicitly set the `data_classification_policy` attribute of the loader.

[Unitxt|INFO|loaders.py:98] 2024-06-23 10:18:42,310 >>
Loading limited to 300 instances by setting unitxt.settings.global_loader_limit;
Generating train split: 300 examples [00:00, 359.59 examples/s]
[Unitxt|INFO|loaders.py:98] 2024-06-23 10:18:43,223 >>
Loading limited to 300 instances by setting unitxt.settings.global_loader_limit;
Generating train split: 300 examples [00:00, 304.19 examples/s]
.[Unitxt|INFO|loaders.py:123] 2024-06-23 10:18:44,277 >> LoadHF sets 'data_classification_policy' to ['public'] by default when loading from Huggingface hub.
To use a different value or remove this message, explicitly set the `data_classification_policy` attribute of the loader.

[Unitxt|INFO|loaders.py:185] 2024-06-23 10:18:48,028 >>
Loading filtered by: lambda instance: instance["language"]=="eng";
[Unitxt|INFO|loaders.py:98] 2024-06-23 10:18:48,031 >>
Loading limited to 300 instances by setting unitxt.settings.global_loader_limit;
Generating train split: 250 examples [00:00, 300.38 examples/s]
[Unitxt|INFO|loaders.py:98] 2024-06-23 10:18:48,955 >>
Loading limited to 300 instances by setting unitxt.settings.global_loader_limit;
Generating train split: 250 examples [00:00, 304.98 examples/s]
.[Unitxt|INFO|loaders.py:123] 2024-06-23 10:18:49,841 >> LoadHF sets 'data_classification_policy' to ['public'] by default when loading from Huggingface hub.
To use a different value or remove this message, explicitly set the `data_classification_policy` attribute of the loader.

[Unitxt|INFO|loaders.py:98] 2024-06-23 10:18:56,568 >>
Loading limited to 300 instances by setting unitxt.settings.global_loader_limit;
Generating train split: 300 examples [00:04, 63.98 examples/s]
.[Unitxt|INFO|loaders.py:123] 2024-06-23 10:19:01,351 >> LoadFromDictionary sets 'data_classification_policy' to ['proprietary'] by default when loading from python dictionary.
To use a different value or remove this message, explicitly set the `data_classification_policy` attribute of the loader.

..[Unitxt|INFO|loaders.py:98] 2024-06-23 10:19:02,105 >>
Loading limited to 300 instances by setting unitxt.settings.global_loader_limit;
Generating train split: 160 examples [00:00, 9094.45 examples/s]
Generating train split: 80 examples [00:00, 9467.69 examples/s]
[Unitxt|INFO|loaders.py:98] 2024-06-23 10:19:02,884 >>
Loading limited to 10 instances by setting LoadFromHFSpace.loader_limit;
Generating train split: 10 examples [00:00, 1991.41 examples/s]
Generating train split: 10 examples [00:00, 1861.07 examples/s]
Generating train split: 1 examples [00:00, 517.18 examples/s]
Generating train split: 1 examples [00:00, 608.49 examples/s]
Generating train split: 2 examples [00:00, 954.23 examples/s]
Generating train split: 2 examples [00:00, 1158.97 examples/s]
Generating train split: 2 examples [00:00, 1044.66 examples/s]
Generating train split: 2 examples [00:00, 1180.00 examples/s]
Generating train split: 1 examples [00:00, 558.05 examples/s]
Generating train split: 1 examples [00:00, 420.52 examples/s]
Generating train split: 2 examples [00:00, 942.86 examples/s]
Generating train split: 2 examples [00:00, 1034.61 examples/s]
Generating train split: 2 examples [00:00, 939.16 examples/s]
Generating train split: 2 examples [00:00, 658.76 examples/s]
Generating train split: 1 examples [00:00, 527.59 examples/s]
Generating train split: 1 examples [00:00, 584.57 examples/s]
Generating train split: 2 examples [00:00, 1089.29 examples/s]
Generating train split: 2 examples [00:00, 1156.09 examples/s]
Generating train split: 2 examples [00:00, 1074.50 examples/s]
Generating train split: 2 examples [00:00, 1115.21 examples/s]
.[Unitxt|INFO|loaders.py:98] 2024-06-23 10:19:03,539 >>
Loading limited to 300 instances by setting unitxt.settings.global_loader_limit;
[Unitxt|INFO|loaders.py:98] 2024-06-23 10:19:03,542 >>
Loading limited to 300 instances by setting unitxt.settings.global_loader_limit;
/home/dafna/workspaces/unitxt/src/unitxt/loaders.py:337: ResourceWarning: unclosed file <_io.TextIOWrapper name='/tmp/tmpa6ngt8mw/train.csv' mode='r' encoding='utf-8'>
  yield row.to_dict()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
[Unitxt|INFO|loaders.py:98] 2024-06-23 10:19:03,548 >>
Loading limited to 300 instances by setting unitxt.settings.global_loader_limit;
[Unitxt|INFO|loaders.py:98] 2024-06-23 10:19:03,552 >>
Loading limited to 300 instances by setting unitxt.settings.global_loader_limit;
/home/dafna/workspaces/unitxt/src/unitxt/loaders.py:337: ResourceWarning: unclosed file <_io.TextIOWrapper name='/tmp/tmpa6ngt8mw/test.csv' mode='r' encoding='utf-8'>
  yield row.to_dict()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
[Unitxt|INFO|loaders.py:123] 2024-06-23 10:19:03,560 >> LoadCSV sets 'data_classification_policy' to ['proprietary'] by default when loading from local files.
To use a different value or remove this message, explicitly set the `data_classification_policy` attribute of the loader.

/home/dafna/workspaces/unitxt/src/unitxt/dataclass.py:336: DeprecationWarning: Field 'use_query' is deprecated. From now on, default behavior is compatible to use_query=True. Please remove this field from your code.
  original_init(self, *args, **kwargs)
[Unitxt|INFO|loaders.py:123] 2024-06-23 10:19:03,564 >> LoadCSV sets 'data_classification_policy' to ['proprietary'] by default when loading from local files.
To use a different value or remove this message, explicitly set the `data_classification_policy` attribute of the loader.

[Unitxt|INFO|loaders.py:98] 2024-06-23 10:19:03,567 >>
Loading limited to 300 instances by setting unitxt.settings.global_loader_limit;
[Unitxt|INFO|loaders.py:98] 2024-06-23 10:19:03,571 >>
Loading limited to 300 instances by setting unitxt.settings.global_loader_limit;
.
----------------------------------------------------------------------
Ran 14 tests in 57.144s

OK
(virtual39) dafna@DESKTOP-GM8R3J7:~/workspaces/unitxt$
```

**Comment by dafnapension** [comment]
@elronbandel , Is the issue still there?

**Comment by dafnapension** [comment]
Resolved by PR #930
